### PR TITLE
Mail persistence

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -13,4 +13,5 @@ django-environ = "==0.4.5"
 django-model-utils = "*"
 django-reversion = "==3.0.3"
 djangorestframework = "==3.9.2"
+psycopg2 = "==2.8.2"
```
Comment (charlesluokb) [inline Pipfile]: what is this for?

File: mail/data_processing.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from mail.dtos import EmailMessageDto
+from mail.helpers import sender_to_source, process_attachment
+from mail.models import LicenseUpdate
+from mail.serializers import LicenseUpdateSerializer, InvalidEmailSerializer
+
+
+def process_and_save_email_message(dto: EmailMessageDto):
+    data = {}
+    last_hmrc_number = LicenseUpdate.objects.last().hmrc_run_number
```
Comment (charlesluokb) [inline mail/data_processing.py]: what does `last` exactly do?

File: mail/data_processing.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    data["hmrc_run_number"] = (
+        last_hmrc_number + 1 if last_hmrc_number != 99999 else 0
+    )  # TODO: Extra logic to generalise
+    data["source_run_number"] = dto.run_number
```
Comment (charlesluokb) [inline mail/data_processing.py]: 👍 like `hmrc_run_number` and `source_run_number`

File: mail/enums.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Temporary values, THESE ARE NOT CORRECT OR FINAL
+
+
+class ReceptionStatusEnum:
+    ACCEPTED = "accepted"
```
Comment (charlesluokb) [inline mail/enums.py]: shall `accepted` be `Accepted`?

File: Pipfile (hunk 1 above)
Comment (cstobart) [reply Pipfile]: It allows the use of postgres

File: mail/data_processing.py (hunk 2 above)
Comment (cstobart) [reply mail/data_processing.py]: returns the last db record

File: mail/enums.py (hunk 4 above)
Comment (cstobart) [reply mail/enums.py]: lower case for enum, sentence case for choices

File: mail/enums.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    choices = [
+        (ACCEPTED, "Accepted"),
```
Comment (charlesluokb) [inline mail/enums.py]: `,` shall be removed?

**Pull request merged**